Firmware/depth-sensing-JSN-SRO4_Gen3_RK-SD-GPS-temp-integration-attempt1

```diff
@@ -0,0 +1,324 @@
+
+SYSTEM_THREAD(ENABLED);
+SYSTEM_MODE(SEMI_AUTOMATIC);
+
+// Declare one of these as a global variable to manage the JSN-SR04 sensor
+JSN_SR04_Gen3 distanceSensor;
+
+
+//===============================================================================
+// INITIALIZATION
+//===============================================================================
+void setup() {
+
+    Serial.begin(115200);
+
+    // Initialize the sensor configuration from setup()
+    distanceSensor
+        .withTrigPin(TRIG_PIN)
+        .withEchoPin(ECHO_PIN)
+        .withUnusedPins(UNUSED_PIN0, UNUSED_PIN3)
+        .withCallback(distanceCallback)
+        .withSamplePeriodic(500ms)
+        .setup();
+
+        // Set up GPS
+        GPS.begin(9600);
+
+        // uncomment next line to turn on RMC (recommended minimum) and GGA (fix data) including altitude
+        GPS.sendCommand(PMTK_SET_NMEA_OUTPUT_RMCGGA);
+
+        // uncomment next line to turn on only the "minimum recommended" data
+        //GPS.sendCommand(PMTK_SET_NMEA_OUTPUT_RMCONLY);
+
+        // Set the update rate
+        GPS.sendCommand(PMTK_SET_NMEA_UPDATE_1HZ); // 1 Hz update rate
+        // For the parsing code to work nicely and have time to sort thru the data, and print it out we don't suggest using anything higher than 1 Hz
+    
+        // Initialize the SD library
+        if (!SD.begin(SD_CHIP_SELECT, SPI_FULL_SPEED)) {
+            Serial.println("failed to open card");
+            return;
+        }
+
+}
+
+
+//===============================================================================
+// MAIN
+//===============================================================================
+void loop() {
+    // You must call this frequently from loop(), preferable on every execution
+    distanceSensor.loop();
+   
+    // read data from the GPS in the 'main loop'
+    char c = GPS.read();
+
+    // if a sentence is received, we can check the checksum, parse it...
+    if (GPS.newNMEAreceived()) {
+        if (!GPS.parse(GPS.lastNMEA())) // this also sets the newNMEAreceived() flag to false
+            return; // we can fail to parse a sentence in which case we should just wait for another
+    }
+
+    // approximately every second or so, print out the current stats
+    if (millis() - timer > 1000) {
+        timer = millis(); // reset the timer
+
+        serialPrintGPSTime();
+
+        temperatureC = getTemp();
+        depth = distanceSensor.distanceCallback();
+
+        // If GPS gets a fix, print out and save good data
+        if (GPS.fix) {  
+            // Blink to let us know you're alive
+            led_state = !led_state;
+            digitalWrite(MY_LED, led_state); // turn the LED on (HIGH is the voltage level)  
+
+            serialPrintGPSLoc();
+
+        }
+
+        printToFile(); 
+
+    }
+
+
+
+/* ---------------------- DISTNACE CALLBACK FUNCTION (printing---------------------- */
+float distanceCallback(JSN_SR04_Gen3::DistanceResult result) {
```

Comment:
This new script looks like it's getting close, but I think this DistanceResult chunk (especially the result object) has to be integrated into functions like printToFile for the distance values to get saved to the SD card. If it's only called here and not used in the SD card logging function, it will be printed to screen but not saved to SD.